src/MongoBson.pas

```diff
@@ -1473,7 +1473,7 @@ function TBsonBuffer.startObject(const Name: UTF8String): Boolean;
 var
   child: Pointer;
 begin
-  child := bson_new;
+  GetMem(child, SizeOf(bson_t));
```

Comment:
@eugene-convey , why not using bson_new() here as it was ?
bson_new() all it does is allocate a bson object and initialize some of its members to zero.
I usually don't like relying on structs declared on user languages of DLL because the code is harder to maintain and brittle making it error prone to alignment problems.

See code of bson_new():

https://github.com/Convey-Compliance/libbson/blob/development/src/bson/bson.c#L1857-L1875